src/ast/variables.ts

```diff
@@ -0,0 +1,70 @@
+import { FieldNode, ValueNode, OperationDefinitionNode } from 'graphql';
+
+import { getName } from './node';
+import { Variables } from '../types';
+
+/** Evaluates a given ValueNode to a JSON value taking vars into account */
+export const evaluateValueNode = (node: ValueNode, vars: Variables) => {
+  switch (node.kind) {
+    case 'NullValue':
+      return null;
+    case 'IntValue':
+      return parseInt(node.value, 10);
+    case 'FloatValue':
+      return parseFloat(node.value);
+    case 'ListValue':
+      return node.values.map(v => evaluateValueNode(v, vars));
+    case 'ObjectValue':
+      return node.fields.reduce((obj, field) => {
+        obj[getName(field)] = evaluateValueNode(field.value, vars);
+        return obj;
+      }, {});
+    case 'Variable':
+      const varValue = vars[getName(node)];
+      return varValue !== undefined ? varValue : null;
+    default:
+      return node.value;
+  }
+};
+
+/** Evaluates a fields arguments taking vars into account */
+export const getFieldArguments = (
+  node: FieldNode,
+  vars: Variables
+): null | Variables => {
+  if (node.arguments === undefined || node.arguments.length === 0) {
+    return null;
+  }
+
+  return node.arguments.reduce((args, arg) => {
+    args[getName(arg)] = evaluateValueNode(arg.value, vars);
+    return args;
+  }, {});
```

Comment:
I messed up, I think this will need to be Object.create(null) to prevent prototype pollution